service/src/event/service.rs

```diff
@@ -121,71 +122,40 @@ impl CeramicEventService {
     }
 
     #[tracing::instrument(skip(self, items), level = tracing::Level::DEBUG, fields(items = items.len()))]
-    pub(crate) async fn insert_events_from_carfiles<'a>(
+    /// This function is used to insert events from a carfile requiring that the history is local to the node.
+    /// This is likely used in API contexts when a user is trying to insert events. Events discovered from
+    /// peers can come in any order and we will discover the prev chain over time. Use
+    /// `insert_events_from_carfiles_remote_history` for that case.
+    pub(crate) async fn insert_events_from_carfiles_local_history<'a>(
         &self,
         items: &[recon::ReconItem<'a, EventId>],
-        must_have_history: bool,
     ) -> Result<recon::InsertResult> {
         if items.is_empty() {
             return Ok(InsertResult::default());
         }
 
-        let mut insert_now = Vec::with_capacity(items.len());
-        let mut notify_task_new = Vec::with_capacity(items.len());
-        let mut insert_after_history_check = if must_have_history {
-            Vec::with_capacity(items.len())
-        } else {
-            Vec::default()
-        };
+        let ordering =
+            InsertEventOrdering::discover_deliverable_local_history(items, &self.pool).await?;
+        self.process_events(ordering).await
+    }
 
-        let mut background_task_deliverable = HashMap::new();
-        for item in items {
-            let cid = item.key.cid().ok_or_else(|| {
-                Error::new_invalid_arg(anyhow::anyhow!("EventID is missing a CID: {}", item.key))
-            })?;
-            // we want to end a conversation if any of the events aren't ceramic events and not store them
-            // this includes making sure the key matched the body cid
-            let (insertable_body, maybe_prev) = Self::parse_event_carfile(cid, item.value).await?;
-            let mut insertable = EventInsertable::try_new(item.key.to_owned(), insertable_body)?;
-            if let Some(meta) = maybe_prev {
-                if must_have_history {
-                    insert_after_history_check.push((meta, insertable));
-                } else {
-                    background_task_deliverable.insert(insertable.body.cid, meta);
-                    insert_now.push(insertable);
-                }
-            } else {
-                insertable.deliverable(true);
-                notify_task_new.push(DeliveredEvent::new(
-                    insertable.body.cid,
-                    insertable.body.cid,
-                ));
-                insert_now.push(insertable);
-            }
+    #[tracing::instrument(skip(self, items), level = tracing::Level::DEBUG, fields(items = items.len()))]
+    pub(crate) async fn insert_events_from_carfiles_remote_history<'a>(
```

Comment:
Can you add a similar comment to this function ? 

```
    /// This function is used to insert events from a carfile NOT requiring that the history is local to the node and expecting to learn the history eventually.
```

Reply:
yep, good call